pkg/sfu/downtrack.go

```diff
@@ -104,22 +120,25 @@ type DownTrack struct {
 	bufferFactory *buffer.Factory
 	payload       *[]byte
 
-	currentSpatialLayer atomicInt32
-	targetSpatialLayer  atomicInt32
+	// translationMu protection start
```

Comment:
One option I considered was to make a structure called `Translator` and manage all the stuff under this lock in that structure rather than spread all over the place in the `DownTrack` structure. Please let me know if that will make things cleaner and I will make the change.

Reply:
I would recommend stopping reviewing if you think the separation will help just thinking about it as the diff is painful to read. I don't want to waste your time with another read if we decide to do the separate structure.

Reply:
agree with  a dedicate structure will make things cleaner and easy to read.

Reply:
Sounds good. Will do the split.

Reply:
@cnderrauber Split it out into a separate struct in this commit - https://github.com/livekit/livekit-server/pull/213/commits/cc2d11f41ab24f972b866d7f7b22de29ea3259b8. Also addressed your other two suggestions.